### ISSUE TITLE
Activation layers give nan as loss value

### ISSUE THREAD

**Issue by inspiralpatterns** [fchollet/keras#6732]
I've been trying to insert batch normalisation inside my convnet.
Doing so, I split between `Conv2D` and `relu`. I therefore deployed something similar:

```
x = Conv2D(256, (3, 3), padding='same', kernel_regularizer=regularizers.l2(0.01), 
           kernel_initializer='he_normal', name='enc')(in)
x = BatchNormalization(axis=-1)(x)
x = Activation('relu')(x)
```

and I always get `nan` values for training and validation loss.
It happens to be the same for the _advanced_ activation layers, such as `LeakyReLU` or `PReLU`.
I'd like to get values instead, especially if I do get values for the metric.
Anyone knows why such a behaviour?

**Comment by leaprovenzano** [comment]
are you using theano or tensoflow backend?  

If theano then batch norm axis should be 1 and that could be the cause of your problems. If you're using tensorflow your axis is the correct ( & default axis) so I'm not sure what the problem would be.
